src/wp-includes/class-wp-term-query.php

```diff
@@ -738,7 +734,8 @@ public function get_terms() {
 
 		if ( empty( $terms ) ) {
 			wp_cache_add( $cache_key, array(), 'terms', DAY_IN_SECONDS );
-			return array();
+			$this->terms = array();
+			return $this->terms;
```

Comment:
The previous behavior was for `$this->terms` to be set to all of the terms, ie an `array()`. With this PR, no terms are returned. However, `$this->terms` is still set to `null`. This is a breaking change _when_ an `array()` is expected, ie when using in an iterator such as `foreach`.

The code change here sets the `terms` property to an empty `array` _before_ returning.